mono/metadata/image.c

```diff
@@ -1932,6 +1932,12 @@ mono_image_close_except_pools_all (MonoImage**images, int image_count)
 	}
 }
 
+static void
+remove_cached_module(gpointer key, gpointer value, gpointer user_data)
+{
+	mono_dl_close((MonoDl*)value);
```

Comment:
Same here are below. Should you `g_free` the key/name?

Reply:
I'd say most definitely! Interesting that mono didn't do that anyway. I'll fix it and retest the code.
Do you think running the full mono test suite should be enough to verify such a change? (Alternative would mean that I bring back the hackweek unity reloading hacks (nothing trunkworthy there yet) and manually check against that...)